### ISSUE TITLE
BUG: ensure_success is not passed down to replaced jobs

### ISSUE THREAD

**Issue by FabiPi3** [materialsproject/jobflow#423]
**Describe the bug**
If you run a job or flow with run_locally, a keyword `ensure_success` is provided. If set to True, an error should be raised if the job/flow execution throws an error. Default is False, which means the error will be catched and reported, but not re-raised.
Now I can define a job, which gets replaced with a new job. If this job throws an exception, jobflow doesn't re-raise this error although `ensure_success` is set to True.

**To Reproduce**
Here is a minimal code example:

```
from jobflow import job, Response, run_locally, Flow

@job
def keks():
    return Response(replace=baum())

@job
def baum():
    raise ValueError()

keks_job = keks()
out = run_locally(keks_job, ensure_success=True)
print(out)
```

I tested the newest development version of jobflow:
Version: 0.1.13.post36+g98f91da
and Python:
Version: 3.11.4

Running this leads to:

```
2023-09-07 21:42:10,305 INFO Started executing jobs locally
2023-09-07 21:42:10,366 INFO Starting job - keks (c9bd6a57-cfaa-41c5-aca4-80a96f513a75)
2023-09-07 21:42:10,591 INFO Finished job - keks (c9bd6a57-cfaa-41c5-aca4-80a96f513a75)
2023-09-07 21:42:10,592 INFO Starting job - baum (c9bd6a57-cfaa-41c5-aca4-80a96f513a75, 2)
2023-09-07 21:42:10,594 INFO baum failed with exception:
Traceback (most recent call last):
  File ".../code/jobflow/src/jobflow/managers/local.py", line 102, in _run_job
    response = job.run(store=store)
               ^^^^^^^^^^^^^^^^^^^^
  File ".../code/jobflow/src/jobflow/core/job.py", line 583, in run
    response = function(*self.function_args, **self.function_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../code/jobflow/test.py", line 9, in baum
    raise ValueError()
ValueError

2023-09-07 21:42:10,594 INFO Finished executing jobs locally
{'c9bd6a57-cfaa-41c5-aca4-80a96f513a75': {1: Response(output=None, detour=None, addition=None, replace=Flow(name='Flow', uuid='4bde8836-db62-44f2-aa79-ab8bfdb99589')
1. Job(name='baum', uuid='c9bd6a57-cfaa-41c5-aca4-80a96f513a75'), stored_data=None, stop_children=False, stop_jobflow=False)}}
```

The flow gets executed, and the print afterwards as well.

**Expected behavior**
As the keyword tells, the success of the run should be ensured and an error should be raised. The print afterwards should not be executed.

**Comment by mcgalcode** [comment]
Hi @FabiPi3, thanks for the report. I made a change that I think addresses your problem in this [branch](https://github.com/materialsproject/jobflow/pull/426). Feel free to pull it down and see if it indeed addresses the issue.

**Comment by FabiPi3** [comment]
> Hi @FabiPi3, thanks for the report. I made a change that I think addresses your problem in this [branch](https://github.com/materialsproject/jobflow/pull/426). Feel free to pull it down and see if it indeed addresses the issue.

Thank you, it indeed solves the described issue. Feel free to mark this as solved.

**Comment by utf** [comment]
Closed in #426
